### PR TITLE
MODINVOICE-39: Implement basic GET /voucher/voucher-lines/<id>

### PR DESCRIPTION
**Pull request by javalkar-varun**
## Purpose

https://issues.folio.org/browse/MODINVOICE-39

## Approach

- Implement GET `/voucher/voucher-lines/<id>` by making a proxy call to `/voucher-storage/voucher-lines`
- Add unit tests

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -109,6 +109,18 @@
         }
       ]
     },
+    {
+      "id": "voucher",
+      "version": "1.0",
+      "handlers": [
+        {
+          "methods": ["GET"],
```
Comment (KVupp) [inline descriptors/ModuleDescriptor-template.json]: can be added to the existing "voucher" interface

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -229,6 +241,24 @@
...
+      "permissionName": "voucher.all",
+      "displayName": "Invoice module - all Voucher permissions",
+      "description": "Entire set of permissions needed to use Vouchers",
+      "subPermissions": [
+        "voucher.vouchers.item.get"
+      ],
+      "visible": false
+    },    
+    {
+      "permissionName": "voucher.all",
```
Comment (KVupp) [inline descriptors/ModuleDescriptor-template.json]: duplicate permissionName, please add it as a sub permission to the existing permission Name

File: src/main/java/org/folio/rest/impl/VouchersImpl.java [hunk 3]
```diff
@@ -2,30 +2,49 @@
...
-  private Void handleErrorResponse(Handler<AsyncResult<Response>> asyncResultHandler, AbstractHelper helper, Throwable t) {
-    asyncResultHandler.handle(succeededFuture(helper.buildErrorResponse(t)));
-    return null;
-
+  
+  @Validate
+  @Override
+  public void getVoucherVoucherLinesById(String id, String lang, Map<String, String> okapiHeaders,
+                                         Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    logger.info("== Get Voucher Line by Id for an existing invoice ==");
```
Comment (KVupp) [inline src/main/java/org/folio/rest/impl/VouchersImpl.java]: possible typo in the log message for invoice

File: src/test/java/org/folio/rest/impl/VoucherLinesApiTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import static org.folio.rest.RestVerticle.OKAPI_HEADER_TENANT;
+import static org.folio.rest.impl.AbstractHelper.ID;
+import static org.junit.Assert.assertEquals;
+
+
+public class VoucherLinesApiTest extends ApiTestBase {
+
+  private static final Logger logger = LoggerFactory.getLogger(VoucherLinesApiTest.class);
+
+  static final Header NON_EXIST_CONFIG_X_OKAPI_TENANT = new Header(OKAPI_HEADER_TENANT, "invoicetest");
```
Comment (KVupp) [inline src/test/java/org/folio/rest/impl/VoucherLinesApiTest.java]: looks like this variable is not being used anywhere?

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (javalkar-varun) [reply descriptors/ModuleDescriptor-template.json]: Done

File: src/test/java/org/folio/rest/impl/VoucherLinesApiTest.java (hunk 4 above)
Comment (javalkar-varun) [reply src/test/java/org/folio/rest/impl/VoucherLinesApiTest.java]: Yes, removed it.

File: src/main/java/org/folio/rest/impl/VouchersImpl.java (hunk 3 above)
Comment (javalkar-varun) [reply src/main/java/org/folio/rest/impl/VouchersImpl.java]: Done

File: descriptors/ModuleDescriptor-template.json (hunk 2 above)
Comment (javalkar-varun) [reply descriptors/ModuleDescriptor-template.json]: Done

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (javalkar-varun) [reply descriptors/ModuleDescriptor-template.json]: Done

**Pull request merged**